config/src/parameters/user.rs

```diff
@@ -387,6 +387,9 @@ impl Sumeragi {
         } = self;
 
         let trusted_peers = construct_unique_vec(trusted_peers.unwrap_or(vec![]))?;
+        for peer in &trusted_peers {
+            assert!(peer.public_key.algorithm() == iroha_crypto::Algorithm::Secp256k1);
```

Comment:
I think we should return human readable error with explanation that only certain public key types are allowed.

Reply:
Should we as well check key type of peer in `Register<Peer>` instruction?